__tests__/ExpensiMark-HTML-test.js

```diff
@@ -609,6 +609,60 @@ test('Test a url ending with a closing parentheses autolinks correctly', () => {
     expect(parser.replace(testString)).toBe(resultString);
 });
 
+test('Test urls autolinks correctly', () => {
+    let testString = 'test@expensify.com https://www.expensify.com\n' +
```

Comment:
This is a little hard to read and review, so I was thinking maybe we should add a test for each of these cases, but I think a better idea would be to make this an array with entries for each case, and I think even better would be an array of arrays, where the test string and the expected string are next to each other for easy comparison, readability, review, and updates.

If you agree, can you update accordingly?

Reply:
How about dividing it into 3~4 test sections according to test string type, like https://github.com/Expensify/App/issues/16762#issuecomment-1598810396

Reply:
That works, but I still arrays are better than string concats, and an array of array such that the test and expected strings are next to each other is better than both

Reply:
The following implementation is acceptable ? 

```
    const testCases = [
        {
            testString: 'test@expensify.com https://www.expensify.com',
            resultString: '<a href="mailto:test@expensify.com">test@expensify.com</a> <a href="https://www.expensify.com" target="_blank" rel="noreferrer noopener">https://www.expensify.com</a>',
        },
        {
            testString: 'test@expensify.com-https://www.expensify.com',
            resultString: '<a href="mailto:test@expensify.com">test@expensify.com</a>-<a href="https://www.expensify.com" target="_blank" rel="noreferrer noopener">https://www.expensify.com</a>',
        },
    ];

    testCases.forEach(testCase => {
        expect(parser.replace(testCase.testString)).toBe(testCase.resultString);
    });
```

Reply:
Yes that looks better than what I had in mind :)